File: frontend/src/api/types/workspace.ts
export interface Workspace {
  id: number;
  name: string;
  active_model: string;
  description: string;
  category: string;
  router_mode: string;
  memory_profile_id: number | null;
  memory_profile_name: string;
  created_at: string;
  updated_at: string;
}

export interface CreateWorkspaceRequest {
  name: string;
  active_model: string;
  description: string;
  category: string;
  router_mode: string;
  memory_profile_id: number | null;
}

export interface SwitchWorkspaceRequest {
  name: string;
}

File: api/schemas/workspace.py
"""Workspace schemas."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class WorkspaceBase(BaseModel):
    name: str
    active_model: str = ""
    description: str = ""
    category: str = "general"
    router_mode: str = "disabled"
    memory_profile_id: int | None = None


class CreateWorkspaceRequest(WorkspaceBase):
    pass


class UpdateWorkspaceRequest(BaseModel):
    name: str | None = None
    active_model: str | None = None
    description: str | None = None
    category: str | None = None
    router_mode: str | None = None


class WorkspaceResponse(WorkspaceBase):
    id: int
    memory_profile_name: str = "Shared"
    created_at: str = ""
    updated_at: str = ""

    model_config = ConfigDict(from_attributes=True)


class SwitchWorkspaceRequest(BaseModel):
    name: str
